libpam_misc/Makefile.am

```diff
@@ -15,7 +15,8 @@ libpam_misc_la_LDFLAGS = -no-undefined -version-info 82:1:82
 if HAVE_VERSIONING
   libpam_misc_la_LDFLAGS += -Wl,--version-script=$(srcdir)/libpam_misc.map
 endif
-libpam_misc_la_LIBADD = $(top_builddir)/libpam/libpam.la
+libpam_misc_la_LIBADD = $(top_builddir)/libpam_internal/libpam_internal.la \
+			$(top_builddir)/libpam/libpam.la
```

Comment:
Another question is whether libpam_internal.la should be listed here and in other places before or after libpam.la.
The rule of thumb is that the library that's used by other libraries should be listed after the libraries which use it.

So if libpam_internal uses libpam, then libpam_internal should be listed before libpam, and vice versa.

Reply:
In general I agree. In this case though, the nature of libpam_internal leads to the effect that it's fully included in libpam, so there is no real inter-dependency going on.

But I'm fine with reordering these lines to have libpam_internal in front of libpam if you want to.

Reply:
Since it won't hurt and it's correct from a logical point of view (even if it's not technically required), I've adjusted the ordering.